core/string/node_path.cpp

```diff
@@ -199,6 +199,21 @@ Vector<StringName> NodePath::get_subnames() const {
 	return Vector<StringName>();
 }
 
+StringName NodePath::get_concatenated_names() const {
+	ERR_FAIL_COND_V(!data, StringName());
+
+	if (!data->concatenated_path) {
+		int pc = data->path.size();
+		String concatenated;
+		const StringName *sn = data->path.ptr();
+		for (int i = 0; i < pc; i++) {
+			concatenated += i == 0 ? sn[i].operator String() : "/" + sn[i];
+		}
+		data->concatenated_path = concatenated;
```

Comment:
Minor issue but `prepend_period` and `simplify` should invalidate the `concatenated_path` cache.
That said, probably not an issue in practice. `prepend_period` is only used in one place in the code.

Reply:
Since `get_concatenated_subnames()` is already implemented in the same way, it may be better to modify it at the same time.

Reply:
Oh, I misread. That might need to be taken into account for self-directories. It may be an issue somewhere else, but certainly not necessary right now.

Reply:
Honestly, I don't think `prepend_period` or `simplify` should be mutating NodePaths. They are allowed to share `data`, and could then be sent between threads (such as in a deferred signal argument). It's really weird to have an immutable data structure which shares references (for good reason), and then have two methods which allow mutation in a thread-unsafe way.
Maybe easiest not to worry about this for now. Sometime, NodePath should be changed to be fully immutable and have these two methods return a new NodePath.